src/components/nft-details/nft-details.tsx

```diff
@@ -164,22 +164,31 @@ export const NftDetails = () => {
                 <TraitsListLoader />
               ) : (
                 <>
-                  {Object.keys(nftDetails.traits).map((key) => (
-                    <NFTTraitsChip
-                      label={getTraitName(key)}
-                      key={key}
-                      name={nftDetails?.traits[`${key}`].name}
-                      rimValue={`${
-                        nftDetails?.traits[`${key}`].occurance
-                      } (${
-                        nftDetails?.traits[`${key}`].rarity &&
-                        roundOffDecimalValue(
+                  {Object.keys(nftDetails.traits).map((key) => {
+                    const occurance = nftDetails?.traits[`${key}`]
+                      .occurance
+                      ? nftDetails?.traits[`${key}`].occurance
+                      : undefined;
+                    const rarity = nftDetails?.traits[`${key}`].rarity
+                      ? roundOffDecimalValue(
                           nftDetails?.traits[`${key}`].rarity,
                           2,
                         )
-                      }%)`}
-                    />
-                  ))}
+                      : undefined;
+
+                    return (
+                      <NFTTraitsChip
+                        label={getTraitName(key)}
+                        key={key}
+                        name={nftDetails?.traits[`${key}`].name}

```

Comment:
On top of everything said before, we could probably get all the desired values upfront, a few lines before at the top:

```
const { name, rarity } = nftDetails?.traits?.[`${key}`];
```

👆 The advantage here, is that we're only testing once for traits?.[key]

Which then we could:

```suggestion
                        name={name}
```